pkg/azureutils/azure_disk_utils.go

```diff
@@ -476,11 +478,19 @@ func ParseDiskParameters(parameters map[string]string) (ManagedDiskParameters, e
 			if deviceSettings, err := optimization.GetDeviceSettingFromAttribute(k); err == nil {
 				diskParams.DeviceSettings[filepath.Join(consts.DummyBlockDevicePathLinux, deviceSettings)] = v
 			} else {
-				return diskParams, fmt.Errorf("invalid parameter %s in storage class", k)
+				invalidParameters = append(invalidParameters, k)
```

Comment:
We should explicitly ignore `azureconstants.RequestedSizeGib` since this parameter is [added by `CreateVolume`](https://github.com/kubernetes-sigs/azuredisk-csi-driver/blob/13b204b5dc205f08bdc7f67449e01b25b401c209/pkg/provisioner/cloudprovisioner.go#L280).

Reply:
Isn't two always going to be a possibility since `Status` cannot be updated on create if we need to create a new `AzVolume` during `Recover` e.g. in upgrade scenarios?

Reply:
We short circuit reconciliation until recovery is complete so as long as we populate and update the status of `AzVolume` during recovery, 2) shouldn't happen.